src/main/java/com/google/sps/servlets/YoutubeServlet.java

```diff
@@ -39,25 +40,28 @@
   // Parameters required by YouTube API to retrieve the comment threads
   private static final String SNIPPET_PARAMETERS = "snippet,replies";
   private static final String ORDER_PARAMETER = "relevance";
-  // TODO: obtain actual API Key
-  private static final String DEVELOPER_KEY = "OUR_API_KEY";
+
   private static final String APPLICATION_NAME = "SAY";
+  private static final String DEVELOPER_KEY = "AIzaSyDYfjWcy1hEe0V7AyaYzgIQm_rT-9XbiGs";
 
   /**
-   * Applies parameters to the YouTube API, then extracts comments. URL is the only true variable;
-   * for this application we will always want order to be relevance and max results to be 100, the
-   * API's arbitrary limit
+   * Retrieves comments from designated URL, passes them off to CommentAnalysis object to be wrapped
+   * into Statistics object, then writes the Statistics object to the frontend.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
-      throws IOException, ServletException {
+      throws ServletException {
     try {
       String url = request.getParameter(URL_PARAMETER);
       CommentThreadListResponse commentResponse = generateYouTubeRequest(url).execute();
-      // TODO: input commentResponse into commentAnalysis class to get useful data
-      Statistics statistics = new Statistics(new ArrayList<Double>());
+
+      CommentAnalysis commentAnalysis = new CommentAnalysis();
+      Statistics statistics = commentAnalysis.computeOverallStats(commentResponse);
+      commentAnalysis.closeLanguage(); // Ensure memory  used by commentAnalysis is cleaned
+
       String json = new Gson().toJson(statistics);
       response.setContentType("application/json");
+      System.out.println(json);
```

Comment:
Is this just for debugging?

I would remove this from the PR and maybe as a later effort use a logging library for this kind of work.

I.e. we could think about adding something like Flogger (https://google.github.io/flogger/)

Reply:
Oops, yes it is, forgot to remove it.